src/test/kotlin/edu/carroll/cs389/service/tracker/trackservice/TrackServiceGeneralPersistenceTest.kt

```diff
@@ -0,0 +1,154 @@
+package edu.carroll.cs389.service.tracker.trackservice
+
+import edu.carroll.cs389.service.tracker.TrackService
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.test.annotation.DirtiesContext
+import org.springframework.test.util.AssertionErrors.assertTrue
+
+@SpringBootTest
+@DirtiesContext(classMode = DirtiesContext.ClassMode.BEFORE_EACH_TEST_METHOD)
+class TrackServiceGeneralPersistenceTest {
+
+    @Autowired
+    private lateinit var trackService: TrackService
+
+    /*****************************************/
+    /* General datastore Persistence Testing */
+    /*****************************************/
+
+    @Test
+    fun datastoreIsEmptyWithoutInsertTest() {
+        assertTrue(
+            "datastoreIsEmptyWithoutInsertTest(): test datastore (newly instantiated and cleaned after every test" +
+                    " should be empty without inserting data",
+            trackService.query().isEmpty()
+        )
+    }
+
+    @Test
+    fun singleInsertTest() {
+        val ipv4: String = "192.168.1.1"
+        val browser: String = "Firefox"
+        val browserMajorVersion: String = "110"
+        val platform: String = "Ubuntu"
+        val platformVersion: String = "22.04 LTS"
+        val requestURI: String = "/"
+
+        assertTrue(
+            "persistenceOfTwoInsertsTest(): should succeed with valid user agent",
+            trackService.trackClient(ipv4, browser, browserMajorVersion, platform, platformVersion, requestURI)
+        )
+
+        assertTrue(
+            "persistenceOfTwoInsertsTest(): inserted two records, should be of size 2",
+            trackService.query().size == 1
+        )
+
+        val user = trackService.query()[0]
```

Comment:
Instead of doing two calls, you should return and cache the query results, then check the size and elements.  Database calls should be minimized, even in unit tests, especially as the application grows in size.